src/main/java/core/basesyntax/Circle.java

```diff
@@ -0,0 +1,33 @@
+package core.basesyntax;
+
+import static java.lang.Math.PI;
+
+public class Circle extends Figure {
+    private int radius;
+
+    public void setRadius(int radius) {
+        this.radius = radius;
+    }
+
+    @Override
+    public String draw() {
+        return "Figure: circle, area: "
+                + getArea()
+                + " sq. units, radius: "
+                + radius
+                + " units, color: "
+                + getColor();
+    }
+
+    @Override
+    public double getArea() {
+        return radius * radius * PI;
+    }
+
+    public Circle prepareCircle() {
+        Circle circle = new Circle();
+        circle.setRadius(side());
+        circle.setColor(color());
+        return circle;
+    }
```

Comment:
The models must only have a state, not a behavior
move this logic to `FigureSupplier`
fix it everywhere
```suggestion
```